### ISSUE TITLE
Terminal component installs the global eyre handler based on color choice

### ISSUE THREAD

**Issue by mzabaluev** [iqlusioninc/abscissa#589]
The Terminal component of the framework [performs initialization](https://github.com/iqlusioninc/abscissa/blob/df52b4daade83c78879a08047396729ca4e2cc84/core/src/terminal/component.rs#L19) of the global eyre handler with `color_eyre::install`, unless the application's color choice is set to `Never`. If the handler installation fails, a panic occurs.

This is problematic, because the application should be in control of setting such things as program-global error/panic handlers.
Or, if it's really decided that the framework is so opinionated that it makes eyre setup part of its functionality, it should provide an API to override it. The colorization choice alone is a poor way to control this, because the application may want to set up error reporting differently based on other settings (e.g. whether backtraces are enabled).

**Comment by tony-iqlusion** [comment]
Agreed this is a problem. In general the quality of the `color-eyre` integration is not great at present.

It would be nice to expose something more configurable. `zebrad` is something I've been looking at for a reference:

https://github.com/ZcashFoundation/zebra/blob/7457edcb86ae24846f55ee1efb4b84f6ddb138e0/zebrad/src/application.rs#L261-L304

cc @yaahc

**Comment by yaahc** [comment]
I'd probably go about this by making sure users of abscissa have a chance to `install` their own Handler before abcissa does automatically, and make that auto installation fail gracefully. It seems like right now you may already have the first element and only need to remove the panic to enable this approach.
